profiling-ffi/src/exporter.rs

```diff
@@ -165,128 +169,125 @@ impl From<RequestBuildResult> for Result<Box<Request>, String> {
     fn from(result: RequestBuildResult) -> Self {
         match result {
             // Safety: Request is opaque, can only be built from Rust.
-            RequestBuildResult::Ok(ok) => Ok(unsafe { Box::from_raw(ok) }),
-            RequestBuildResult::Err(err) => Err(String::from(err)),
+            RequestBuildResult::Ok(ok) => Ok(unsafe { Box::from_raw(ok.as_ptr()) }),
+            RequestBuildResult::Err(err) => Err(err.to_string()),
         }
     }
 }
 
 /// If successful, builds a `ddog_prof_Exporter_Request` object based on the
 /// profile data supplied. If unsuccessful, it returns an error message.
 ///
-/// The main use of the `ddog_prof_Exporter_Request` object is to be used in
-/// `ddog_prof_Exporter_send`, which will take ownership of the object. Do not
-/// pass it to `ddog_prof_Exporter_Request_drop` after that!
-///
 /// # Safety
 /// The `exporter`, `additional_stats`, and `endpoint_stats` args should be
 /// valid objects created by this module, except NULL is allowed.
 #[no_mangle]
 #[must_use]
 pub unsafe extern "C" fn ddog_prof_Exporter_Request_build(
-    exporter: Option<NonNull<ProfileExporter>>,
+    exporter: *mut ProfileExporter,
     start: Timespec,
     end: Timespec,
     files: Slice<File>,
-    additional_tags: Option<&ddcommon_ffi::Vec<Tag>>,
-    endpoints_stats: Option<&profiled_endpoints::ProfiledEndpointsStats>,
+    additional_tags: *const ddcommon_ffi::Vec<Tag>,
+    endpoints_stats: *const profiled_endpoints::ProfiledEndpointsStats,
```

Comment:
Can these still be `NULL` now that they're not `Option`? I don't quite understand the trade-off in this change :sweat_smile:

Reply:
Yes.